diesel/src/query_builder/sql_query.rs

```diff
@@ -76,36 +77,53 @@ impl SqlQuery {
     pub fn bind<ST, Value>(self, value: Value) -> UncheckedBind<Self, Value, ST> {
         UncheckedBind::new(self, value)
     }
+
+    /// Internally boxes future calls on `bind` and `sql` so that they don't
+    /// change the type.
+    ///
+    /// This allows doing things you otherwise couldn't do, e.g. `bind`ing in a
+    /// loop.
+    pub fn into_boxed<'f, DB: Backend>(self) -> BoxedSqlQuery<'f, DB, Self> {
+        BoxedSqlQuery::new(self)
+    }
+
+    /// Appends a piece of SQL code at the end.
+    pub fn sql<T: Into<String>>(self, sql: T) -> SqlQuery<Self> {
+        SqlQuery::new(self, sql.into())
```

Comment:
Why do we create a new wrapper type here and not just append to sql string of the existing one, similar to `BoxedSqlQuery`?

Reply:
No reason, fixed.